### PR TITLE
add W&I + LOCNESS dataset (BEA-2019 workshop shared task on GEC)

### PR DESCRIPTION
**Pull request by aseifert**
- **Name:** W&I + LOCNESS dataset (from the BEA-2019 workshop shared task on GEC)
- **Description:** https://www.cl.cam.ac.uk/research/nl/bea2019st/#data
- **Paper:** https://www.aclweb.org/anthology/W19-4406/
- **Motivation:** This is a recent dataset (actually two in one) for grammatical error correction and is used for benchmarking in this field of NLP.

### Checkbox

- [x] Create the dataset script `/datasets/my_dataset/my_dataset.py` using the template
- [x] Fill the `_DESCRIPTION` and `_CITATION` variables
- [x] Implement `_infos()`, `_split_generators()` and `_generate_examples()`
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is filled with the different configurations of the dataset and that the `BUILDER_CONFIG_CLASS` is specified if there is a custom config class.
- [x] Generate the metadata file `dataset_infos.json` for all configurations
- [x] Generate the dummy data `dummy_data.zip` files to have the dataset script tested and that they don't weigh too much (<50KB)
- [x] Add the dataset card `README.md` using the template : fill the tags and the various paragraphs
- [x] Both tests for the real data and the dummy data pass.

### CODE REVIEW

File: datasets/wi_locness/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/wi_locness/README.md:2]:
```suggestion
```

File: datasets/wi_locness/wi_locness.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+_HOMEPAGE = "https://www.cl.cam.ac.uk/research/nl/bea2019st/#data"
+
+_LICENSE = ""
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://www.cl.cam.ac.uk/research/nl/bea2019st/data/wi+locness_v2.1.bea19.tar.gz"
+
+
+class WriteImproveLocness(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/wi_locness/wi_locness.py:71]:
```suggestion
class WiLocness(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `wi_locness`

you'll have to regenerate the dataset_infos.json file after this change

**Review by lhoestq — COMMENTED** [review]
Looking good so far thanks !

I left a few comments

Moreover it looks like this PR includes changes about many other files than the ones for your dataset, can you create another branch and another PR please ?

File: datasets/wi_locness/README.md (hunk 1 above)
Comment (aseifert) [reply datasets/wi_locness/README.md:2]: oh shoot, sorry for that! i created a clean PR here: https://github.com/huggingface/datasets/pull/1449

Comment (aseifert) [reply datasets/wi_locness/README.md:2]: you can close this one

**Comment by aseifert** [comment]
I created a clean PR where I also incorporated the suggested changes here: https://github.com/huggingface/datasets/pull/1449

**Pull request closed without merging**